core/aws_ddk_core/resources/_utils.py

```diff
@@ -0,0 +1,84 @@
+import logging
+import platform
+from typing import Optional
+
+import aws_cdk.aws_lambda as lmbda
+import boto3
+from constructs import Construct
+
+AWS_SDK_PANDAS_ARTIFACT_ACCOUNT_ID = "336392948345"
+MAX_VERSION_POLL = 50
+LAYER_PREFIXES = ["AWSDataWrangler", "AWSSDKPandas"]
+logging.basicConfig()
+logger = logging.getLogger("aws-sdk-pandas-layer-lookup")
+
+
+def _get_python_version() -> str:
+    return "Python" + "".join(platform.python_version().split(".")[0:2])
+
+
+def _latest_layer(boto3_client: boto3.client, region: str = "us-east-1", scan_buffer_range: int = 25) -> str:
+    layer_arn = (
+        f"arn:aws:lambda:{region}:{AWS_SDK_PANDAS_ARTIFACT_ACCOUNT_ID}:layer:AWSSDKPandas-{_get_python_version()}"
+    )
+    logger.debug(f" Layer Arn: {layer_arn}")
+    logger.debug(" Scanning versions for latest...")
+    latest = False
+    version = 1
+    while not latest:
```

Comment:
I think we should use [CDK runtime contexts](https://docs.aws.amazon.com/cdk/v2/guide/context.html) for this as well. This way, CDK synth will always synthesize to the same CloudFormation template, unless the synth is explicitly told to ignore the cache in `cdk.context.json`.

You generally want CDK synth to generate the same exact template based on the contents of what's in version control. Runtime contexts are a way to bridge this gap with the idea that sometimes, you do want something to be inferred on synthesis time.

Reply:
Another good reason to do this is to avoid breaking changes. When we release version 3.0 of Wrangler, it will just show up as yet another layer version, but it will be a major release with some breaking changes. By using the CDK runtime context, we'll ensure that we CDK synth doesn't just implicitly upgrade the version, thus potentially breaking code for the customer.

Reply:
@LeonLuttenberger do you mean to be able to pull the specific layer version from runtime context / ddk config instead of lookup?

Reply:
Yes, exactly. This way two consecutive runs of _cdk synth_ will have the same output.